packages/network/src/index.ts

```diff
@@ -1,3 +1,19 @@
-type Fetch = typeof fetch;
+import type { BuildFetchOptions, Fetch, Middleware } from "./types";
 
-export type Middleware = (request: Request, fetch: Fetch) => ReturnType<Fetch>;
+import { Request, fetch } from 'cross-fetch';
+
+export function buildFetch(middleware: Middleware[], options?: BuildFetchOptions): Fetch {
+    return (input: RequestInfo, init?: RequestInit) => {
+        const request = new Request(input, init);
+
+        async function chain(request: Request, middleware: Middleware[]): Promise<Response> {
+            if (middleware.length === 0) return fetch(request)
+
+            return middleware[0](request, (input: RequestInfo, init?: RequestInit) => chain(request, middleware.slice(1)))
+        }
+
+        return chain(request, middleware)
+    }
+}
+
+export * from './types';
```

Comment:
I think we should be careful about stuff like this.  Two points.

1. The mental model should be that everything exported in `index.ts` is public api (including types).

I'd **much** rather we be very explicit about the public API and not accidentally leak an internal type.

2. What's the advantage of putting these types in a separate file anyway?  What I'd sooner have us do is move the implementation of `buildFetch` somewhere else (if this file got large or something) and have it re-exported here, again with the mental model being "${package}/index.ts exports all the public api"

Reply:
good point, I have moved the types to be in the main fetch file.